airframe/src/main/scala/wvlet/airframe/LifeCycleManager.scala

```diff
@@ -256,3 +295,30 @@ object FILOLifeCycleHookExecutor extends LifeCycleEventHandler with LogSupport {
     }
   }
 }
+
+/**
+  * If an injected class implements close() (in AutoCloseable interface), add a shutdown hook to call
+  * close() if there is no other shutdown hooks
+  */
+object CloseableLifeCycleHookFinder extends LifeCycleEventHandler with LogSupport {
+
+  class CloseHook(val injectee: Injectee) extends LifeCycleHook {
+    override def toString: String = s"CloseHook for [${surface}]"
+    override def execute: Unit = {
+      injectee.injectee match {
+        case c: AutoCloseable =>
+          c.close()
```

Comment:
If an exception occurs in lifecycle hooks, how does Airframe work?

Reply:
For now, it will just throw the exception at here:
https://github.com/wvlet/airframe/blob/4112e950fd48072213491abfc329bba344d18ebc/airframe/src/main/scala/wvlet/airframe/LifeCycleManager.scala#L81

So the other shutdown hooks will not be called. It would be good to add an error handler so that we can customize how to handle such exceptions.

Reply:
Added a new ticket for handling exceptions #649